server/src/routes/rootRouter.js

```diff
@@ -46,17 +46,13 @@ rootRouter.get('/logout', (req, res) => {
 
 rootRouter.get('/settings', async (req, res) => {
   try {
-    if (!config.discord.enabled) {
+    if (config.enabledAuthMethods.length === 0 || config.alwaysEnabledPerms.length > 0) {
       req.session.perms = {}
-      Object.keys(config.discord.perms).forEach(perm => req.session.perms[perm] = config.discord.perms[perm].enabled)
-      req.session.save()
-    } else if (config.alwaysEnabledPerms.length > 0) {
-      req.session.perms = {}
-      config.alwaysEnabledPerms.forEach(perm => req.session.perms[perm] = config.discord.perms[perm].enabled)
```

Comment:
This is important to be synced with what is enabled/disabled in discord perm section. If needed, add enabled keys to local/custom auth fields as well.

Reply:
I think the way you had it before it was never really an `alwaysEnabled` : 
- if discord auth is disabled default perms are set according to Discord section settings : if you don’t want to use discord you don’t want to have to set anything in that section at all so pulling default perms should be done from this section. The way I made it `alwaysEnabled` acts as default perms if there is not auth method used at all. 
- then you had test case if `alwaysEnabled` array isn’t empty you compare it again to discord section perms and if there one is false then it’s not always enabled.